ADD codecoverage analysis

patkan opened the pull request:
- uses codecov.io
- codecoverage is analyzed on travis-ci.org and then evaluated on codecov.io
- add testsuite that executes all tests

patkan committed:
ADD codecoverage analysis

 * uses codecov.io
 * codecoverage is analyzed on travis-ci.org and then evaluated on codecov.io
 * add testsuite that executes all tests

mgmax commented:
!["test coverage" ...](https://cloud.githubusercontent.com/assets/1520891/10860411/ed451404-7f67-11e5-95d3-a42ef9650784.jpg)

patkan commented:
:-P :+1:

patkan committed:
add quantifiedcode-issues-badge

Quantified-code is a tool similar to landscape but with an other source of check patterns.
This adds the QC-badge in order to fasten access to the report.
